FirebaseAuth/Sources/Swift/Utilities/AuthErrorUtils.swift

```diff
@@ -0,0 +1,540 @@
+        }
+        if let underlyingError {
+            userInfo[NSUnderlyingErrorKey] = underlyingError
+        }
+        return error(code: .internal(.unexpectedErrorResponse), userInfo: userInfo.isEmpty ? nil : userInfo)
+    }
+
+    @objc public static func malformedJWTError(token: String, underlyingError: Error?) -> Error {
+        var userInfo: [String: Any] = [
+            NSLocalizedDescriptionKey: kFIRAuthErrorMessageMalformedJWT,
+            FIRAuthErrorUserInfoDataKey: token
+        ]
+        if let underlyingError {
+            userInfo[NSUnderlyingErrorKey] = underlyingError
+        }
+        return error(code: .malformedJWT, userInfo: userInfo)
+    }
+
+    @objc public static func RPCResponseDecodingError(deserializedResponse: Any?, underlyingError: Error?) -> Error {
+        var userInfo: [String: Any] = [:]
+        if let deserializedResponse {
+            userInfo[FIRAuthErrorUserInfoDeserializedResponseKey] =  deserializedResponse
+        }
+        if let underlyingError {
+            userInfo[NSUnderlyingErrorKey] = underlyingError
+        }
+        return error(code: .internal(.RPCResponseDecodingError), userInfo: userInfo)
+
+    }
+
+    @objc public static func accountExistsWithDifferentCredentialError(email: String?, updatedCredential: AuthCredential?) -> Error {
+        var userInfo: [String: Any] = [:]
+        if let email {
+            userInfo[FIRAuthErrorUserInfoEmailKey] = email
+        }
+        if let updatedCredential {
+            userInfo[FIRAuthErrorUserInfoUpdatedCredentialKey] = updatedCredential
+        }
+        return error(code: .accountExistsWithDifferentCredential, userInfo: userInfo)
+    }
+
+    @objc public static func blockingCloudFunctionServerResponse(message: String?) -> Error {
+        guard let message else {
+            return self.error(code: .blockingCloudFunctionError, message: message)
+        }
+        var jsonString = message.replacingOccurrences(of: "HTTP Cloud Function returned an error:", with: "")
+        jsonString = jsonString.trimmingCharacters(in: .whitespaces)
+        let jsonData = jsonString.data(using: .utf8) ?? Data()
+        do {
+            let jsonDict = try JSONSerialization.jsonObject(with: jsonData, options: []) as? [String: Any] ?? [:]
+            let errorDict = jsonDict["error"] as? [String: Any] ?? [:]
+            let errorMessage = errorDict["message"] as? String
+            return self.error(code: .blockingCloudFunctionError, message: errorMessage)
+        } catch {
+            return self.JSONSerializationError(underlyingError: error)
+        }
+    }
+
+#if os(iOS)
+
+    @objc public static func secondFactorRequiredError(resolver: MultiFactorResolverWrapper?) -> Error {
+        var userInfo: [String: Any] = [:]
+
+        if let resolver {
+            userInfo[FIRAuthErrorUserInfoMultiFactorResolverKey] = resolver
+        }
+
+        return error(code: .secondFactorRequired, userInfo: userInfo)
+    }
+
```

Comment:
Hi @mortenbekditlevsen,  I'm fixing the build for iOS. [This line](https://github.com/firebase/firebase-ios-sdk/blob/b9762b438c38c05e0ffbfb324319eed2a4a43090/FirebaseAuth/Sources/Backend/FIRAuthBackend.m#L777) is failing to build on iOS. Based on the Swift signature here, I'm not seeing a direct translation. What do you think?

Reply:
Should the API look something like?
```suggestion
secondFactorRequiredErrorWithPendingCredential:response.MFAPendingCredential
//                                                                hints
    @objc public static func secondFactorRequiredError(
      pendingCredential: // ...
      hints: // ...
    ) -> Error {
        var userInfo: [String: Any] = [:]

        if let resolver {
            userInfo[FIRAuthErrorUserInfoMultiFactorResolverKey] = resolver
        }

        return error(code: .secondFactorRequired, userInfo: userInfo)
    }

```

Reply:
Yeah, sorry, this looks like a work in progress. I hope to be able to build using Xcode at some point, so I can build the iOS-specific code to.

I'm not completely certain about the visibility of the  `FIRMultiFactorResolver`, but with the obj-c import it should be something like this:

```
  #if os(iOS)

    @objc public static func secondFactorRequiredError(pendingCredential: String?, hints: [MultiFactorInfo]?)
      -> Error {
      var userInfo: [String: Any] = [:]

      if let pendingCredential, let hints {
          let resolver = FIRMultiFactorResolver(MFAPendingCredential: pendingCredential, hints: hints)
          userInfo[FIRAuthErrorUserInfoMultiFactorResolverKey] = resolver
      }

      return error(code: .secondFactorRequired, userInfo: userInfo)
    }
  #endif
```

Would you like a fix on this branch or on the `auth-swift` one?

Reply:
Ahh, got it. Makes sense. I'm working on #10755 right now which will allow us to build within Xcode when building from CocoaPods. That PR tries to expose non-public headers to the Swift part of the pod. 

Locally, I have been iterating on it and in the process have the library mostly building for iOS. And I'll open a PR after #10755 merges with those fixes to get iOS building. You can push to that branch when ready–– I'll let you know!